modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/AbstractPageMemoryMvPartitionStorage.java

```diff
@@ -962,27 +962,18 @@ public T map(VersionChain treeRow) {
     }
 
     @Override
-    public @Nullable GcEntry peek(HybridTimestamp lowWatermark) {
-        assert THREAD_LOCAL_LOCKER.get() != null;
-
-        // Assertion above guarantees that we're in "runConsistently" closure.
-        throwExceptionIfStorageNotInRunnableState();
-
-        GcRowVersion head = renewableState.gcQueue().getFirst();
-
-        // Garbage collection queue is empty.
-        if (head == null) {
-            return null;
-        }
-
-        HybridTimestamp rowTimestamp = head.getTimestamp();
+    public List<GcEntry> peek(HybridTimestamp lowWatermark, int count) {
+        return busy(() -> {
+            throwExceptionIfStorageNotInRunnableState();
 
-        // There are no versions in the garbage collection queue before watermark.
-        if (rowTimestamp.compareTo(lowWatermark) > 0) {
-            return null;
-        }
+            if (count <= 0) {
+                return List.of();
+            } else if (count == 1) {
+                return peekSingleGcEntryBusy(lowWatermark);
+            }
 
-        return head;
+            return peekGcEntriesBusy(lowWatermark, count);
```

Comment:
Why is it necessary to have a dedicated method for the case of a single entry? Could you please clarify that with a comment?